### PR TITLE
fix: bool conversion

### PR DESCRIPTION
**Pull request by stackchain**

### CODE REVIEW

File: src/auth/authOS.ts [hunk 1]
```diff
@@ -166,7 +167,7 @@ const disableAllEasyConfirmation = () =>
           walletUpdates.push([walletPath, {...wallet, isEasyConfirmationEnabled: false}])
         }
       }
-      return [metaUpdates, walletUpdates]
+      return Promise.resolve([metaUpdates, walletUpdates])
```
Comment (stackchain) [inline src/auth/authOS.ts:170]: clean up

File: src/auth/authOS.ts [hunk 2]
```diff
@@ -183,7 +184,7 @@ export const migrateAuthSetting = async (storage: Storage) => {
   const isLegacyAuth = authSetting == null && !isFirstRun
   if (!isLegacyAuth) return
 
-  const isAuthWithOS = await storage.getItem(OLD_OS_AUTH_KEY).then(Boolean)
+  const isAuthWithOS = await storage.getItem(OLD_OS_AUTH_KEY).then(parseBoolean)
```
Comment (stackchain) [inline src/auth/authOS.ts:187]: `isSystemAuth` JSON stringfied -> '"false"' -> true. migrate to parseBoolean.

**Review by banklesss — APPROVED** [review]

File: src/auth/authOS.test.ts [hunk 3]
```diff
@@ -43,10 +43,22 @@ describe('migrateAuthSetting', () => {
   })
 
   it('old store is pin, method = "pin"', async () => {
-    await storage.setItem(ENCRYPTED_PIN_HASH_KEY, JSON.stringify('encrypted-hash'))
+    await storage.multiSet([
+      [ENCRYPTED_PIN_HASH_KEY, JSON.stringify('encrypted-hash')],
+      [OLD_OS_AUTH_KEY, JSON.stringify(false)],
```
Comment (stackchain) [inline src/auth/authOS.test.ts:48]: the storage prior to the migrations run, was not represented correctly

File: src/hooks/index.ts [hunk 4]
```diff
@@ -961,7 +961,7 @@ export const useAuthSetting = (storage: Storage, options?: UseQueryOptions<AuthS
 export const getAuthSetting = async (storage: Storage) => {
   const authSetting = parseAuthSetting(await storage.getItem(AUTH_SETTINGS_KEY))
   if (isAuthSetting(authSetting)) return authSetting
-  return Promise.reject(new Error('useAuthSetting invalid data'))
+  return Promise.reject(new Error('getAuthSetting invalid data'))
```
Comment (stackchain) [inline src/hooks/index.ts:964]: clean up

**Review by wolverineks — APPROVED** [review]

**Pull request merged**